k4FWCore/components/UniqueIDGenSvc.cpp

```diff
@@ -11,18 +11,22 @@ StatusCode UniqueIDGenSvc::initialize() {
 }
 
 const size_t bits32 = std::numeric_limits<uint32_t>::digits;
+const size_t bits64 = std::numeric_limits<uint64_t>::digits;
 const size_t bitsSizeT = std::numeric_limits<size_t>::digits;
 
 size_t UniqueIDGenSvc::getUniqueID(
-  uint32_t evt_num, uint32_t run_num, std::string name) const 
+  int64_t seed, uint32_t evt_num, uint32_t run_num, std::string name) const
 {
-  std::bitset<bits32> event_num_bits(evt_num);
-  std::bitset<bits32> run_num_bits(run_num);
+  std::bitset<bits64> seed_bits(seed);
```

Comment:
This has to be like this
```suggestion
  std::bitset<bits64> seed_bits(this->seed);
```
And the service should have a seed property, and the function signature should not change.